fe/fe-core/src/main/java/org/apache/doris/nereids/properties/FuncDeps.java

```diff
@@ -78,6 +82,32 @@ public boolean isEmpty() {
         return items.isEmpty();
     }
 
+    private void dfs(Set<Slot> parent, Set<Set<Slot>> visited, Set<FuncDepsItem> circleItem) {
```

Comment:
Seems that current circleItem depends on traversing order, and if the useful item is removed by chance, will we can't do the cooresponding opt at that time?